messaging/src/main/java/io/micronaut/messaging/annotation/MessageBody.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.messaging.annotation;
+
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import io.micronaut.context.annotation.AliasFor;
+import io.micronaut.core.bind.annotation.Bindable;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+/**
+ * An annotation that can be applied to method argument to indicate that the method argument is bound from message body.
+ *
+ * @author Graeme Rocher
+ * @since 1.0
+ */
+@Documented
+@Retention(RUNTIME)
+@Target({ElementType.PARAMETER})
+@Bindable
+public @interface MessageBody {
```

Comment:
I don't think this is backwards compatible for use cases like https://github.com/micronaut-projects/micronaut-rabbitmq/blob/7a41c432cc57c1a0974b507faff7b293328ca638/rabbitmq/src/main/java/io/micronaut/rabbitmq/bind/RabbitBinderRegistry.java#L76. If someone switched to using this with RabbitMQ the annotation type would be `MessageBody` and there is no binder setup for that. It seems either we do a big bang for 3.0, or introduce the new annotations now but do not deprecate the existing ones. We can then update the modules to support both and then deprecate the old ones for 2.5, with removal in 3.0.

Reply:
There is an alias for on the annotation so it should activate the `@Body` annotation as well and not break anything, the alternative is an annotation transformer to maintain compatibility

Reply:
The alias only applies if the value is set, and its common to not set the value for `@Body`. `getAnnotationTypeByStereotype` would choose the first annotation, so even in the case where the value is set, it's unclear which annotation it would pick.

Reply:
yeah hence why I think a annotation transformer would solve that. Will write a test

Reply:
@jameskleeh I added a transformer to make the annotations backwards compatible now. Can you check and merge if you agree?